src/pm_data_translator.jl

```diff
@@ -1,398 +1,166 @@
+    return pm_map
 end
 
-function get_device_to_pm(
-    ix::Int,
-    branch::PSY.PhaseShiftingTransformer,
-    device_formulation::Type{StaticBranchUnbounded},
-)
-    PM_branch = Dict{String, Any}(
-        "br_r" => PSY.get_r(branch),
-        "shift" => PSY.get_α(branch),
-        "br_x" => PSY.get_x(branch),
-        "g_to" => 0.0,
-        "g_fr" => 0.0,
-        "b_fr" => PSY.get_primary_shunt(branch) / 2,
-        "f_bus" => PSY.get_number(PSY.get_arc(branch).from),
-        "br_status" => Float64(PSY.get_available(branch)),
-        "t_bus" => PSY.get_number(PSY.get_arc(branch).to),
-        "b_to" => PSY.get_primary_shunt(branch) / 2,
-        "index" => ix,
-        "angmin" => -π / 2,
-        "angmax" => π / 2,
-        "transformer" => true,
-        "tap" => PSY.get_tap(branch),
-    )
-    return PM_branch
-end
+function get_pm_map(sys::PSY.System, component_type::Type{T}) where {T <: PSY.Component}
+    devices = PSY.get_components(component_type, sys)
+    PM_devices = Dict{String, component_type}()
+    for (ix, device) in enumerate(devices)
+        PM_devices["$(ix)"] = device
+    end
 
-function get_device_to_pm(
-    ix::Int,
-    branch::PSY.Transformer2W,
-    device_formulation::Type{D},
-) where {D <: AbstractBranchFormulation}
-    PM_branch = Dict{String, Any}(
-        "br_r" => PSY.get_r(branch),
-        "rate_a" => PSY.get_rate(branch),
-        "shift" => 0.0,
-        "rate_b" => PSY.get_rate(branch),
-        "br_x" => PSY.get_x(branch),
-        "rate_c" => PSY.get_rate(branch),
-        "g_to" => 0.0,
-        "g_fr" => 0.0,
-        "b_fr" => PSY.get_primary_shunt(branch) / 2,
-        "f_bus" => PSY.get_number(PSY.get_arc(branch).from),
-        "br_status" => Float64(PSY.get_available(branch)),
-        "t_bus" => PSY.get_number(PSY.get_arc(branch).to),
-        "b_to" => PSY.get_primary_shunt(branch) / 2,
-        "index" => ix,
-        "angmin" => -π / 2,
-        "angmax" => π / 2,
-        "transformer" => true,
-        "tap" => 1.0,
-    )
-    return PM_branch
+    return PM_devices
 end
 
-function get_gen_to_pm(
-    ix::Int,
-    gen::PSY.ThermalGen,
-)
-    PM_branch = Dict{String, Any}(
-        "br_r" => PSY.get_r(branch),
-        "rate_a" => PSY.get_rate(branch),
-        "shift" => 0.0,
-        "rate_b" => PSY.get_rate(branch),
-        "br_x" => PSY.get_x(branch),
-        "rate_c" => PSY.get_rate(branch),
-        "g_to" => 0.0,
-        "g_fr" => 0.0,
-        "b_fr" => PSY.get_primary_shunt(branch) / 2,
-        "f_bus" => PSY.get_number(PSY.get_arc(branch).from),
-        "br_status" => Float64(PSY.get_available(branch)),
-        "t_bus" => PSY.get_number(PSY.get_arc(branch).to),
-        "b_to" => PSY.get_primary_shunt(branch) / 2,
-        "index" => ix,
-        "angmin" => -π / 2,
-        "angmax" => π / 2,
-        "transformer" => true,
-        "tap" => 1.0,
-    )
-    return PM_branch
+function get_components_to_pm(
```

Comment:
I don't understand the difference between this function and the one above. They seem to do the same thing.

Reply:
`get_components_to_pm` creates the PM dict entries for each type. The PM dict entries don't guarantee that you can map back to a particular `Component`. So I created the pm_map thing that creates a `Dict{String, Component}` where the keys are the same as the keys in the PM dict so that you can get the component associated with a PM dict entry.